thetis/configuration.py

```diff
@@ -195,7 +195,38 @@ def info(self):
         return result + super(PairedEnum, self).info()
 
 
-class FrozenHasTraits(HasTraits):
+class OptionsBase(object):
+    """Abstract base class for all options classes"""
+
+    def name(self):
```

Comment:
I'd like to make this an `@abstractproperty` but if I use `_metaclass__ = ABCMeta` I get metaclass conflict in `FrozenHasTraits` class:
```
TypeError: metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases
```
Some magic to work around it? @wence- ?